src/main/java/org/jfrog/hudson/util/BuildUniqueIdentifierHelper.java

```diff
@@ -139,8 +142,24 @@ public static String getBuildNumber(Run build) {
 
         return buildNumber;
     }
+    
+    public static String getCustomBuildNameResolved(String customBuildName, EnvVars env) {
+    	return Util.replaceMacro(customBuildName, env);
+    }
 
+    public static String getBuildNameConsiderOverride(BuildInfoAwareConfigurator configurator, Run build, EnvVars env) {
+        return configurator.isOverrideBuildName() ? getCustomBuildNameResolved(configurator.getCustomBuildName(), env) : BuildUniqueIdentifierHelper.getBuildName(build);
+    }
+    
     public static String getBuildNameConsiderOverride(BuildInfoAwareConfigurator configurator, Run build) {
```

Comment:
I'm concerned about leaving this signature of the *getBuildNameConsiderOverride* method. This could cause bugs, because usages of this APIs may get an unresolved build name.
I believe all usages can be modified to use the new API.
BTW, *build.getEnvironment()* seems to be deprecated.

Reply:
As soon as I find the time, I'll try a second attempt. So then I would 
* replace the changed function anywhere where I can find it
* use the TaskListener in order to resolve build environments
* if I'm not able to get the environment map, I would pass null to my function
* in my function I will test if the Override BuildName even has symbols for variables in it
* only if thats the case and I was able to resolve the build environment I will change the build name